### PR TITLE
Add osu!-styled colour picker control

### PR DESCRIPTION
**Pull request by bdach**
Design is 99% outdated at this point but I just want to try and start moving things forward toward a conclusion as I've been literally trying to get editor colour selection done for months.

2021-06-30-232634_2560x1018_scrot

Not sure on the `OsuColourPicker` naming. Nothing in the `UserInterfaceV2` namespace has the `Osu` class name prefix, but not having it was conflicting with framework names in a horribly ugly manner. And from what I can see `UserInterfaceV2` is already partway to being obsolete itself anyway...

### CODE REVIEW

File: osu.Game/Graphics/UserInterfaceV2/OsuHSVColourPicker.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        }
+
+        private class OsuSaturationValueSelector : SaturationValueSelector
+        {
+            public OsuSaturationValueSelector()
+            {
+                SelectionArea.CornerRadius = 10;
+                SelectionArea.Masking = true;
+                // purposefully use hard non-AA'd masking to avoid edge artifacts.
+                SelectionArea.MaskingSmoothness = 0;
```
Comment (bdach) [inline osu.Game/Graphics/UserInterfaceV2/OsuHSVColourPicker.cs:67]: The following is what happens if this is omitted (magnified 3x):

2021-06-30-233138_108x100_scrot

Unsure if there is a better way of getting rid.

**Comment by peppy** [comment]
@bdach to save me hours of searching, where is the design document you're using as reference for this?

**Comment by peppy** [comment]
The triangle selection "nub" needs to be changed, I think. It's currently really ugly, bad UX, and doesn't actually work as a marker on the edges:

20210701 154824 (dotnet)
can't actually be dragged

20210701 154919 (dotnet)
doesn't work as a marker (can't tell where it's pointing)

20210701 154953 (dotnet)
is ugly

@arflyte without changing the overall design to you have a proposal to replace the arrow here?

**Comment by bdach** [comment]
The design in this pull was based off the images in #2434, still.

The new public figma library has an entirely different design of the setup screen, but that one is missing everything but the most basic components. So not sure in general how to proceed best to minimise wasted effort.

**Comment by bdach** [comment]
I won't claim that I have any design skills but I can offer the following alternative "nub":

2021-07-02-000950_1482x601_scrot

Also reduced the corner radius on the hue slider a touch, but didn't delete it entirely as it'd start looking wrong in the exact opposite way.

**Comment by peppy** [comment]
Seems fine, let's go with that for now.

**Review by peppy — APPROVED** [review]

**Pull request merged**